content/BCPatterns/generic-method-pattern/index.md

```diff
@@ -0,0 +1,327 @@
+  - one line (functioncall) after an IF-clause
+  - one line (functioncall) after a repeat
+- readable function calls
+- ...
+
+**Call out to the method from a table**
+
+Now, there is a reason that the one global procedure in the method-codeunit is `internal`.  We shouldn't be calling this procedure directly.  We should be calling it through it's "class" (usually, a table can be seen as a class).  That means, we would have to create a public (not internal) procedure on table-level, which can be used in the business logic.
+
+In our example above, this could be that table extension:
+
+```AL
+tableextension 53100 "Customer Ext BASE" extends Customer
+{
+    procedure BlockCustomer(HideDialog: Boolean)
+    var
+        WLDBlockCustomerMeth: Codeunit "WLD BlockCustomer Meth";
+    begin
+        WLDBlockCustomerMeth.BlockCustomer(Rec, HideDialog);
+    end;
+
+}
+```
+
+This practice improves readability.  In fact, by doing this, you just extended the suggesions-list in VSCode (IntelliSense) indicating a new method that your class can do.  This is very convenient for the developer that might need your new method.
+
+*Note - it could very well be that there simply isn't any table that can act as a class for our method.  In that case, you could use a codeunit as well.*
+
+**Naming Conventions**
+
+You might have noticed that the naming of our method is quite strict:
+- codeunit name: `WLD BlockCustomer Meth`
+- internal proc: `BlockCustomer`
+- do-procedure: `DoBlockCustomer`
+
+It is important to align these namings.  It indicates that the codeunit only does one thing (remember: encapsulation), and it improves searchability from outside the codeunit (for example when you're searching symbols or something).
+
+## Usage: 
+
+TODO
+
+## Benefits:
+
+As I said, it will facilitate a lot of advantages.  Let's explain a bit more in depth:
+
+### Extensibility
+Thanks to the *event layer*, by applying this pattern for all methods, we will automatically have the bare minimum of events that we need to hook into a method: the `OnBefore-` and the `OnAfter`.  Of course it would make sense to even add more events to the method when it would make sense (eg, when you're inserting a record in a table, it might be interesting to also raise an event just before you call the insert).
+
+### Decouplability
+Thanks to these same events, and the fact the pattern foresees a handler as well, we are able to "decouple" our method as well.  What do I mean with that?  Well, we can simply subscribe to the `OnBefore`event, and set `handled` to `true`. This means it will never execute the do-procedure, which means, the original procedure/method/business logic is "decoupled".
+
+We can use this obviously for implementing our own method (a new way to accomplish this method), or the disable the method by simply subscribing to it, and only providing the `handled := true` in our subscriber.  However, there are many more usages where we can use this for.  
```

Comment:
"or **the** disable the method by simply subscribing to it" Replace first "the" with "to"